### PR TITLE
First attempt at using prisma to connect to a database

### PR DESCRIPTION
**Pull request by ewels**
Maybe not the right way to go. Happy to close if an alternative is better.

### CODE REVIEW

**Review by ewels — COMMENTED** [review]
Do you think it's worth thinking about making this somehow run async, so that it returns the response immediately and then continues with the DB log on the backend afterwards? Or is that overkill?

File: src/pages/version.php.ts [hunk 1]
```diff
@@ -1,6 +1,95 @@
...
+      console.error(e);
+      await prisma.$disconnect();
+      process.exit(1);
+    });
+}
+
 export const get: APIRoute = ({ params, request }) => {
+  const version = new URL(request.url).searchParams.get("v");
+  if (!version) {
+    return new Response("No version provided", {
```
[the remark this answers is not in the document]
Comment (ewels) [reply src/pages/version.php.ts:71]: Can log it in a dedicated category then? Then easy to exclude..

[the remark this answers is not in the document]
Comment (mashehu) [reply src/pages/version.php.ts]: like this https://github.com/ewels/MultiQC_website/pull/41/commits/dab5e5d19b86b04e4236403c9185ab6e70d52fb5 ?

**Comment by ewels** [comment]
Got an extremely useful build error log:

```
You incorrectly pinned it to debian-openssl-1.1.x
This probably happens, because you built Prisma Client on a different platform.
(Prisma Client looked in "/opt/build/repo/node_modules/@prisma/client/runtime/libquery_engine-debian-openssl-1.1.x.so.node")
Searched Locations:
  /opt/build/repo/node_modules/.prisma/client
  /opt/build/repo/node_modules/@prisma/client
  /opt/build/repo/node_modules/@prisma/client
  /opt/build/repo/node_modules/.prisma/client
  /opt/build/repo
  /tmp/prisma-engines
  /opt/build/repo/node_modules/.prisma/client
To solve this problem, add the platform "debian-openssl-1.1.x" to the "binaryTargets" attribute in the "generator" block in the "schema.prisma" file:
Failed during stage 'building site': Build script returned non-zero exit code: 2 (https://ntl.fyi/exit-code-2)
generator client {
  provider      = "prisma-client-js"
  binaryTargets = ["native"]
}
Then run "prisma generate" for your changes to take effect.
Read more about deploying Prisma Client: https://pris.ly/d/client-generator
```

**Comment by mashehu** [comment]
Did you follow the setup I linked to on slack?

**Comment by ewels** [comment]
Yup!

**Comment by ewels** [comment]
PR deployments are all failing I think because of the cache? When I trigger from latest commit on the branch without cache they seem to work..

**Pull request merged**